core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/WorkflowTools.java

```diff
@@ -0,0 +1,334 @@
+              consequence.getValue());
+
+      // if no consequence is verified, then last one will be used
+      if (consequence.getKill()) {
+        for (final String state : states) {
+          removeAffectations(instance, model.getState(state));
+        }
+
+        // process external actions
+        processTriggers(consequence, instance, event);
+
+        return true;
+      }
+      State[] targetStates = consequence.getTargetStates();
+      State[] unsetStates = consequence.getUnsetStates();
+
+      // for each target state, set eligibility to true
+      for (final State targetState : targetStates) {
+        if (targetState != null) {
+          String name = targetState.getName();
+          eligibleStates.put(name, name);
+        }
+      }
+
+      // for each unset state, set eligibility to false
+      for (final State unsetState : unsetStates) {
+        if (unsetState != null) {
+          String name = unsetState.getName();
+          eligibleStates.remove(name);
+        }
+      }
+
+      // notify user
+      List<QualifiedUsers> notifiedUsersList = consequence.getNotifiedUsers();
+      for (QualifiedUsers notifiedUsers : notifiedUsersList) {
+        Actor[] actors = instance.getActors(notifiedUsers, null);
+        Task[] tasks = taskManager.createTasks(actors, instance);
+        String message;
+
+        for (int i = 0; i < actors.length; i++) {
+          message = notifiedUsers.getMessage();
+          if (message == null || message.length() == 0) {
+            message = action.getDescription("", "");
+          }
+
+          // check if sender has been hardcoded in the model
+          String senderId = notifiedUsers.getSenderId();
+          User forcedUser = null;
+          if (senderId != null) {
+            try {
+              forcedUser = userManager.getUser(senderId);
+            } catch (WorkflowException we) {
+              SilverLogger.getLogger(WorkflowTools.class)
+                  .error("Error while processing {0}: impossible to find the sender with id {1}",
+                      event.getActionName(), senderId);
+            }
+          }
+
+          User sender = (forcedUser == null) ? event.getUser() : forcedUser;
+          taskManager.notifyActor(tasks[i], sender, actors[i].getUser(), message);
+        }
+      }
+    } catch (Exception e) {
+      // change the action status of the step : Process Failed
+      step.setActionStatus(-1);
+      instance.updateHistoryStep(step);
+      throw new WorkflowException("WorkflowEngineThread.process",
+          "workflowEngine.EX_ERR_PROCESS_EVENT", e);
+    }
+
+    // change the action status of the step
+    step.setActionStatus(1); // Processed
+    instance.updateHistoryStep(step);
+
+    // Compute states and affectations
+    computeStates(instance, event, eligibleStates, oldActiveStates);
+
+    // change the action status of the step
+    // Affectations done
+    step.setActionStatus(2);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)